### PR TITLE
DOC: fix SA05 errors in docstrings for pandas.core.window Expanding and Rolling, and SeriesGroupBy

### LINKED ISSUE #57392: DOC: fix SA05 errors in docstrings
Pandas has a script for validating docstrings:

https://github.com/pandas-dev/pandas/blob/1d7aedcd19470f689e4db625271802eb49531e80/ci/code_checks.sh#L145-L206

Currently, some methods fail the SA05 check.

The task here is:

- take 2-4 methods
- run: scripts/validate_docstrings.py --format=actions --errors=SA05 `method-name`
- check if validation docstrings passes for those methods, and if it’s necessary fix the docstrings according to whatever error is reported
- remove those methods from code_checks.sh
- commit, push, open pull request
- Please don't comment `take` as multiple people can work on this issue. You also don't need to ask for permission to work on this, just comment on which methods are you going to work.

If you're new contributor, please check the [contributing guide](https://pandas.pydata.org/docs/dev/development/contributing.html)

### PR DESCRIPTION
**Pull request by williamking1221**
- [x] closes #57392 
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

File: ci/code_checks.sh [hunk 1]
```diff
@@ -3182,14 +3182,6 @@ if [[ -z "$CHECK" || "$CHECK" == "docstrings" ]]; then
         pandas.util.hash_pandas_object # There should be no backslash in the final line, please keep this comment in the last ignored function
     RET=$(($RET + $?)) ; echo $MSG "DONE"
 
-    MSG='Partially validate docstrings (SA05)' ;  echo $MSG
-    $BASE_DIR/scripts/validate_docstrings.py --format=actions --errors=SA05 --ignore_functions \
```
Comment (mroeschke) [inline ci/code_checks.sh:3186]: SA05 needs to be added to the list of codes above that validate all docstrings

Comment (williamking1221) [reply ci/code_checks.sh:3186]: To clarify, I'll just need to change line 3186 to 
`$BASE_DIR/scripts/validate_docstrings.py --format=actions --errors=SA05`

File: ci/code_checks.sh [hunk 2]
```diff
@@ -3182,14 +3182,6 @@ if [[ -z "$CHECK" || "$CHECK" == "docstrings" ]]; then
         pandas.util.hash_pandas_object # There should be no backslash in the final line, please keep this comment in the last ignored function
     RET=$(($RET + $?)) ; echo $MSG "DONE"
 
-    MSG='Partially validate docstrings (SA05)' ;  echo $MSG
```
[the remark this answers is not in the document]
Comment (williamking1221) [reply ci/code_checks.sh:3185]: Ah gotcha. Thanks!

[the remark this answers is not in the document]
Comment (jordan-d-murphy) [reply ci/code_checks.sh:3185]: sweet! sorry I think when i added my review it didn't become visible to you for some reason. Sorry for any confusion.

[the remark this answers is not in the document]
Comment (williamking1221) [reply ci/code_checks.sh:3185]: Fixed. Thanks!

**Review by jordan-d-murphy — APPROVED** [review]
lgtm!

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @williamking1221  and @jordan-d-murphy  for the review

**Pull request merged**